packages/muze-utils/src/DataSystem/selection.js

```diff
@@ -42,61 +26,54 @@ class Selection {
      * @return {Selection} Modified selection.
      * @memberof Selection
      */
-    data (newData, idGetter) {
-        if (idGetter) {
-            this._data = [];
-            this._idGetter = idGetter;
-            const tempMap = {};
+    data (newData) {
+        const entryData = new Map();
+        const exitdata = new Map();
+        const tempMap = new Map();
+        const duplicateData = new Map();
 
+        if (!this._idGetter) {
+           // when id-getter is not given
```

Comment:
When no id method is present, then it should consider index as id

Reply:
done